### PR TITLE
Remove drag and drop on kanban and replace it by button for group actions

### PR DESCRIPTION
**Pull request by davidZIVI**
- Fix wrong view on create contracts

- Add colors to m2one fields parrent_id and child_id

- Fix order groups for sds state

### CODE REVIEW

File: sponsorship_compassion/view/contract_view.xml [hunk 1]
```diff
@@ -77,7 +77,7 @@
                         </group>
                         <group>
                             <field name="child_id" domain="[('state', 'in', ('N', 'D', 'I', 'R', 'Z')), ('project_id.suspension', 'not in', ('suspended','fund-suspended'))]"
-                                   options="{'create': false, 'm2o_dialog': false, 'create_edit': false}"/>
+                                   options="{'create': false, 'm2o_dialog': false, 'create_edit': false, 'search_more':True, 'field_color':'state', 'colors':{'R':'red', 'F':'gray', 'X':'gray', 'N':'blue'} }"/>
```
Comment (ecino) [inline sponsorship_compassion/view/contract_view.xml]: You can remove F and X colors as they are filtered in the domain and will never appear in selection.

File: sponsorship_tracking/model/contracts.py [hunk 2]
```diff
@@ -16,29 +16,26 @@
...
-
-        state_transitions = [
-            {start: 'start', end: 'active', signal: 'mail_sent'},
-            {start: 'waiting_welcome', end: 'active', signal: 'welcome_sent'},
-            {start: 'sub_waiting', end: 'no_sub', signal: 'no_sub'},
-        ]
+    def button_mail_sent(self, cr, uid, value, context=None):
+        contract_ids = self.search(
+            cr, uid, [('sds_state', '=', value)], context)
+        for contract in self.browse(cr, uid, contract_ids, context):
```
Comment (ecino) [inline sponsorship_tracking/model/contracts.py]: Browse method is useless, can be replaced by
```for contract_id in contract_ids```

File: sponsorship_tracking/model/contracts.py [hunk 3]
```diff
@@ -16,29 +16,26 @@
...
-            if (state_transition[start] == old_state and
-                    state_transition[end] == new_state):
-                trans_method = getattr(self, state_transition[signal])
-                return trans_method(cr, uid, id, context)
-        else:
-            return False
+    def button_project_mail_sent(self, cr, uid, value, context=None):
+        contract_ids = self.search(
+            cr, uid, [('project_state', '=', value)], context)
+        for contract in self.browse(cr, uid, contract_ids, context):
```
Comment (ecino) [inline sponsorship_tracking/model/contracts.py]: Browse method is useless.

File: sponsorship_tracking/view/contract_view.xml [hunk 4]
```diff
@@ -9,17 +9,19 @@
...
 			<field name="name">follow.contract.compassion.kanban</field>
 			<field name="model">recurring.contract</field>
+            <field name="options">{'test':'test'}</field>
 			<field name="arch" type="xml">
-                <kanban default_group_by="sds_state" create='false' edit='true'>
+                <kanban default_group_by="sds_state" create='false' edit='false'>
                     <field name="name"/>
                     <field name="color"/>
-                    <field name="sds_state"/>
+                    <field name="sds_state" options="{'test':'test'}"/>
```
Comment (ecino) [inline sponsorship_tracking/view/contract_view.xml]: Please remove this test.

File: sponsorship_tracking/view/contract_view.xml [hunk 5]
```diff
@@ -9,17 +9,19 @@
...
                     <field name="state" />
                     <templates>
                         <field name="child_id"/>
                         <t t-name="kanban-box">
                             <div t-attf-class="oe_kanban_color_#{kanban_getcolor(record.color.raw_value)} oe_kanban_card oe_kanban_global_click">
                                 <div class="oe_dropdown_toggle oe_dropdown_kanban">
+                                    <span class="oe_e">í</span>
```
Comment (ecino) [inline sponsorship_tracking/view/contract_view.xml]: What is this "i" ?

File: sponsorship_tracking/view/contract_view.xml [hunk 6]
```diff
@@ -46,7 +48,7 @@
         </record>
 
         <record id="view_tracking_contract_form" model="ir.ui.view">
-			<field name="name">follow.contract.compassion.form</field>
+			<field name="name">recurring.contract.compassion.form</field>
```
Comment (ecino) [inline sponsorship_tracking/view/contract_view.xml]: Not sure if this is a good renaming. It should not be a name already defined in a parent module (sponsorship_compassion).

**Comment by davidZIVI** [comment]
The 'i'  is just for the arrow (transformed by css)

**Comment by ecino** [comment]
:+1:

**Pull request merged**